### DISCUSSION TITLE
Audio Autoplay and Play issue

### DISCUSSION THREAD

**Discussion by CVxTz** [zauberzeug/nicegui#2858] [Q&A] [answered]
### Question

Hello ! 

I am wondering how the autoplay feature works.

```py
from nicegui import ui, Client

@ui.page("/")
async def index(client: Client):

    a = ui.audio('https://cdn.pixabay.com/download/audio/2022/02/22/audio_d1718ab41b.mp3', autoplay=True)
    a.on('ended', lambda _: ui.notify('Audio playback completed'))

    ui.button('Play', on_click=a.play)

ui.run()
```
When I run the example above, the audio does not play on its own when I load the page even with `autoplay=True`. Do you know why ?
Other question,  when I add `a.play()` at the end, it does not play the audio either. The Play button works as expected.

I feel like I am missing the logic on how Nicegui works :/ 

Thanks !

**Comment by falkoschindler** [comment, marked as answer]
Hi @CVxTz,

This is a really confusing issue. But it seems like browsers are blocking autoplay by default.

Here is how you can enable it in firefox, for example:
<img width="498" alt="Screenshot 2024-04-09 at 07 31 47" src="https://github.com/zauberzeug/nicegui/assets/5767091/95513f9f-7f95-4efd-9819-0182ebedd597">

**Reply by CVxTz** [reply]
Thank you !
